src/test/test-models/bad/for_loops/assign_to_loop_var1.stan

```diff
@@ -0,0 +1,14 @@
+functions {
+  int foo(int a) {
+    int vs[2];
+    int y;
+    for (v in vs) v = 3;
```

Comment:
Presumably, it's unnecessary to do this for all three kinds of loops seeing that the logic is precisely the same? Would we want those extra tests going forward, in case they diverge?

Reply:
Thanks for adding those files! I guess the one for simple for loops over range is still missing, but I'm sure that's ok.